plugins/contact/src/index.ts

```diff
@@ -166,5 +169,70 @@ export default plugin(contactId, {
   },
   app: {
     Contacts: '' as Ref<Doc>
+  },
+  string: {
+    PersonAlreadyExists: '' as IntlString
   }
 })
+
+export default contactPlugin
+
+/**
+ * @public
+ */
+export async function findPerson (client: Client, person: Data<Person>, channels: AttachedData<Channel>[]): Promise<FindResult<Person>> {
+  if (channels.length === 0 || person.name.length === 0) {
+    return []
+  }
+  // Take only first part of first name for match.
+  const values = channels.map(it => it.value)
+
+  // Same name persons
+
+  const potentialChannels = await client.findAll(contactPlugin.class.Channel, { value: { $in: values } })
+  let potentialPersonIds = Array.from(new Set(potentialChannels.map(it => it.attachedTo as Ref<Person>)).values())
+
+  if (potentialPersonIds.length === 0) {
+    const firstName = getFirstName(person.name).split(' ').shift() ?? ''
+    const lastName = getLastName(person.name)
+    // try match using just first/last name
+    potentialPersonIds = (await client.findAll(contactPlugin.class.Person, { name: { $like: `${lastName}%${firstName}%` } })).map(it => it._id)
+    if (potentialPersonIds.length === 0) {
+      return []
+    }
+  }
+
+  const potentialPersons: FindResult<Person> = await client.findAll(contactPlugin.class.Person, { _id: { $in: potentialPersonIds } }, {
+    lookup: {
+      _id: {
+        channels: contactPlugin.class.Channel
+      }
+    }
+  })
+
+  const result: FindResult<Person> = []
+
+  for (const c of potentialPersons) {
+    let matches = 0
+    if (c.name === person.name) {
+      matches++
+    }
+    if (c.city === person.city) {
```

Comment:
I'm not sure about city

Reply:
I think just name is not enough. We could tune this later.